enhancements/rhcos/split-rhcos-into-layers.md

```diff
@@ -0,0 +1,514 @@
+  in order to bring up the temporary control plane. This will be adapted so that
+  the installer first live-applies the node image, as if the node did a system
+  update.
+  - Note this work is already in progress:
+    https://github.com/openshift/installer/pull/8742
+  Other components will also need to be adapted (such as assisted-installer, but
+  likely others too).
+- Work with the teams of the affected components to adapt them.
+- Open an openshift/installer PR which updates the bootimages. This will allow
+  us to run more extensive CI tests on the PR.
+- Raise awareness of this change to the OCP org to invite more testing. 
+- Once all issues have been resolved, merge the PR. If new regressions are
+  encountered, revert the PR and fix them.
+- Once satisfied with the new bootimages, the RHCOS pipeline can stop building
+  the old bootimages containing OpenShift components.
+
+##### Phase 2: Switching `rhel-coreos` in CI
+
+- Set up a new job in openshift/release which builds the layered OpenShift node image on
+  top of the bootable container image from the new stream variant. This job should trigger
+  whenever openshift/os changes.
+- Make the CoreOS pipeline also trigger this job whenever it pushes new base images since
+  currently CI doesn't rebuild on changed base images
+- Open MCO and DTK PRs pointing to this layered image in CI (these are the only two components
+  that explicitly reference `rhel-coreos`) as a way to verify CI is happy.
+    - NOTE: There shouldn't be any major CI failures here. One key thing to understand is that
+      we're changing *how* the node image is built, but not *what* its final contents are from
+      a squashed image perspective. So this is mostly a sanity-check that they are indeed
+      equivalent.
+- Once any issues have been resolved, make the new job push to the `:rhel-coreos` tag of the
+  CI imagestream. This simultaneously requires disabling ART's job which syncs RHCOS to that
+  same tag.
+
+##### Phase 3: Switching `rhel-coreos` in production
+
+- Work with CRT to update the OCP release page to better handle the new layered approach.
+    - It currently links to diffs in the RHCOS release browser. But now the
+      release browser will only contain the RHEL diff, not the OCP diff in
+      the node layer. Additionally, if temporary overrides are present in the
+      node image for base packages like the kernel, we really want the release
+      controller to be aware of this. Overall, the RPM diff should always be
+      computed on the final node image. How to implement this remains to be
+      discussed but we will need to close this gap before moving prod. (See
+      "Open Questions" below).
+- Work with ART to adjust how they determine if the RHCOS images are up to date.
+  Currently, they expect both RHEL and OCP content in those images, but that content will
+  be split across separate images now.
```

Comment:
wouldn't the node image built on top of the base image have all the contents?

Reply:
It would. I'm not sure on the details there of how the ART machinery works, but my understanding is that RHCOS builds currently get triggered as part of a larger pipeline that also builds e.g. the kubelet RPM. But with this change, there's no point in rebuilding RHCOS if just OCP packages changed. Instead, a new layered build should happen.